plugin/forward/metrics.go

```diff
@@ -28,6 +28,12 @@ var (
 		Buckets:   plugin.TimeBuckets,
 		Help:      "Histogram of the time each request took.",
 	}, []string{"to", "rcode"})
+	RequestFailures = promauto.NewCounterVec(prometheus.CounterOpts{
+		Namespace: plugin.Namespace,
+		Subsystem: "forward",
+		Name:      "connect_failures_total",
+		Help:      "Counter of connection failures per remote IP address.",
+	}, []string{"to", "proto", "type"})
```

Comment:
`type` is too generic a label name.

But in this case it seems to be the error message. This is probably not a label we want here, as it is likely to generate too much label cardinality.

Please remove the error code from the labels.

Reply:
Perhaps we could do our own categorization of error types into a small number of values (2-4). Without the error type, I think this ends up being redundant with other existing metrics.

Reply:
Yea, I would be more comfortable with a stronger filter.